test/test_system_configuration.py

```diff
@@ -498,3 +536,296 @@ def test_osp_system_structure():
     obj_ref.from_xml(xml_str)
     dict_xml_ref = obj_ref.to_dict_xml()
     assertEqual(dict_xml, dict_xml_ref)
+
+
+def test_adding_updating_deleting_a_initial_values():
+    # create from a dict_xml
+    obj = OspSystemStructure(xml_source=PATH_TO_TEST_SYSTEM_STRUCTURE)
+
+    # Test failure by adding with a wrong component name
+    init_value = OspInitialValue(variable='new_variable', value=10.0)
+    with pytest.raises(TypeError):
+        obj.add_update_initial_value(
+            component_name='This is not a correct name',
+            init_value=init_value
+        )
+
+    # Test adding a new variable
```

Comment:
Consider splitting each of the commented blocks into separate tests or subTests using test class

Reply:
Most of tests are linked with data or setup. I need to learn how to use fixture. Or I could use TestCase class in unittest package.